src/endpoint/endpoint.js

```diff
@@ -107,6 +108,7 @@ async function main(options = {}) {
         process.on('warning', e => dbg.warn(e.stack));
 
         let internal_rpc_client;
+        const object_io = new ObjectIO(location_info);
```

Comment:
I wouldn't move the creation of `object_io`. you can declare it here and create it in the same location as before (under `if (!init_request_sdk)`). for the semaphore monitor, you can ignore the object_io if not defined